drivers/sensor/bmp388/bmp388.h

```diff
@@ -0,0 +1,208 @@
+#define BMP388_REG_FIFO_DATA    0x14
+#define BMP388_REG_FIFO_WTM0    0x15
+#define BMP388_REG_FIFO_WTM1    0x16
+#define BMP388_REG_FIFO_CONFIG1 0x17
+#define BMP388_REG_FIFO_CONFIG2 0x18
+#define BMP388_REG_INT_CTRL     0x19
+#define BMP388_REG_IF_CONF      0x1A
+#define BMP388_REG_PWR_CTRL     0x1B
+#define BMP388_REG_OSR          0x1C
+#define BMP388_REG_ODR          0x1D
+#define BMP388_REG_CONFIG       0x1F
+#define BMP388_REG_CALIB0       0x31
+#define BMP388_REG_CMD          0x7E
+
+/* BMP388_REG_CHIPID */
+#define BMP388_CHIP_ID 0x50
+
+/* BMP388_REG_STATUS */
+#define BMP388_STATUS_FATAL_ERR  BIT(0)
+#define BMP388_STATUS_CMD_ERR    BIT(1)
+#define BMP388_STATUS_CONF_ERR   BIT(2)
+#define BMP388_STATUS_CMD_RDY    BIT(4)
+#define BMP388_STATUS_DRDY_PRESS BIT(5)
+#define BMP388_STATUS_DRDY_TEMP  BIT(6)
+
+/* BMP388_REG_INT_CTRL */
+#define BMP388_INT_CTRL_DRDY_EN_POS  6
+#define BMP388_INT_CTRL_DRDY_EN_MASK BIT(6)
+
+/* BMP388_REG_PWR_CTRL */
+#define BMP388_PWR_CTRL_PRESS_EN    BIT(0)
+#define BMP388_PWR_CTRL_TEMP_EN     BIT(1)
+#define BMP388_PWR_CTRL_MODE_POS    4
+#define BMP388_PWR_CTRL_MODE_MASK   (0x03 << BMP388_PWR_CTRL_MODE_POS)
+#define BMP388_PWR_CTRL_MODE_SLEEP  (0x00 << BMP388_PWR_CTRL_MODE_POS)
+#define BMP388_PWR_CTRL_MODE_FORCED (0x01 << BMP388_PWR_CTRL_MODE_POS)
+#define BMP388_PWR_CTRL_MODE_NORMAL (0x03 << BMP388_PWR_CTRL_MODE_POS)
+
+/* BMP388_REG_OSR */
+#define BMP388_ODR_POS  0
+#define BMP388_ODR_MASK 0x1F
+
+/* BMP388_REG_ODR */
+#define BMP388_OSR_PRESSURE_POS  0
+#define BMP388_OSR_PRESSURE_MASK (0x07 << BMP388_OSR_PRESSURE_POS)
+#define BMP388_OSR_TEMP_POS      3
+#define BMP388_OSR_TEMP_MASK     (0x07 << BMP388_OSR_TEMP_POS)
+
+/* BMP388_REG_CONFIG */
+#define BMP388_IIR_FILTER_POS  1
+#define BMP388_IIR_FILTER_MASK (0x7 << BMP388_IIR_FILTER_POS)
+
+/* BMP388_REG_CMD */
+#define BMP388_CMD_FIFO_FLUSH 0xB0
+#define BMP388_CMD_SOFT_RESET 0xB6
+
+/* default PWR_CTRL settings */
+#define BMP388_PWR_CTRL_ON	    \
+	(BMP388_PWR_CTRL_PRESS_EN | \
+	 BMP388_PWR_CTRL_TEMP_EN |  \
+	 BMP388_PWR_CTRL_MODE_NORMAL)
+#define BMP388_PWR_CTRL_OFF 0
+
+#define BMP388_SAMPLE_BUFFER_SIZE (6)
+
+struct bmp388_cal_data {
+	uint16_t t1;
+	uint16_t t2;
+	int8_t t3;
+	int16_t p1;
+	int16_t p2;
+	int8_t p3;
+	int8_t p4;
+	uint16_t p5;
+	uint16_t p6;
+	int8_t p7;
+	int8_t p8;
+	int16_t p9;
+	int8_t p10;
+	int8_t p11;
+} __packed;
+
+struct bmp388_sample {
+	uint32_t press;
+	uint32_t raw_temp;
+	int64_t comp_temp;
+};
+
+struct bmp388_io_ops {
+	int (*read)(const struct device *dev,
+		    uint8_t reg,
+		    void *data,
+		    size_t length);
+	int (*byte_read)(const struct device *dev,
+			 uint8_t reg,
+			 uint8_t *byte);
+	int (*byte_write)(const struct device *dev,
+			  uint8_t reg,
+			  uint8_t byte);
+	int (*reg_field_update)(const struct device *dev,
+				uint8_t reg,
+				uint8_t mask,
+				uint8_t val);
+};
+
+struct bmp388_config {
+	const struct device *bus;
+	const struct bmp388_io_ops *ops;
+	union {
```

Comment:
I was just suggested by @mbolivar-nordic to not use union here, just to avoid to lose a lot of space in I2C case. See (if you want) what I have done instead in #33506 (two separate structures)

Reply:
No, it is correct. In my case was different. Sorry for noise.